src/providers/ogr/qgsogrprovider.cpp

```diff
@@ -950,8 +950,8 @@ void QgsOgrProvider::loadFields()
     );
     // Set default value for fid, this is needed because
     // the attribute form will not accept a NULL value, passing
-    // -1 will delegate to the back-end.
-    mDefaultValues.insert( 0, QStringLiteral( "-1" ) );
+    // NULL will delegate to the back-end.
+    mDefaultValues.insert( 0, QStringLiteral( "NULL" ) );
```

Comment:
This is what will be shown on the form, right? Could this be `tr( "Autogenerate" )`?

Reply:
Yes, I thought about that, no strong opinions, but I feel that a `NULL` is more appropriate here (see the full comment in the commit) https://github.com/qgis/QGIS/pull/5679/commits/1c9655fb39058881c520438f3dc72835ce30d700

Reply:
I mean, `NULL` is certainly more technical ... but it's what you normally pump into  DB record when you want the default applied. `Autogenerate` may suggest that QGIS has some client-side logic to generate the PK ...

Reply:
> but it's what you normally pump into DB record when you want the default applied.

At least on the DBs I know if you do an insert and explicitly specify NULL it will insert a NULL, not a default value. You have to skip the value or specify the default value expression to trigger the default.